### DISCUSSION TITLE
Lxqt-panel won't hide after fullscreen window apps

### DISCUSSION THREAD

**Discussion by SannUlya** [lxqt/lxqt#2262] [Q&A] [answered]
Hello there.

I am use lxqt-panel, version 1.1.0-1, stable branch on archlinux.
So here the problem. Previously there was no problem anything about lxqt-panel. But, don't know why I got this problem.
When I try to fullscreen the app by click `alt + F11,` previously the lxqt-panel will hide themself in back of the applications window. But now when I try to fullscreen the app. The lxqt-panel won't hide, that still showed up in front of application window. 

I don't know how can I get this problem. Previously I installed lxqt-panel-git, after I got this problem, I try to reinstall lxqt-panel and reset the configuration but seems like didn't fix anything.

Before fullscreen the app:
![04 07 2022_01 08 02_screenshot](https://user-images.githubusercontent.com/98253949/177052343-98402f00-2e2b-4f63-99a6-14af4e7b7ea0.png)

After fullscreen the app:
![04 07 2022_01 08 15_screenshot](https://user-images.githubusercontent.com/98253949/177052349-9b30af94-980b-4f18-9595-bf6024eb4caa.png)

**Comment by tsujan** [comment]
Which app, and which window manager?

**Reply by SannUlya** [reply]
All applications. Example: qterminal, pcmanfm-qt, firefox, and gzdoom

**Reply by tsujan** [reply]
Check your WM. Was it upgraded before the problem started? Have you changed its settings?

You can also test with another WM.

**Reply by SannUlya** [reply]
> Cek WM-mu. Apakah itu ditingkatkan sebelum masalah dimulai? Apakah Anda mengubah pengaturannya?
> 
> Anda juga dapat menguji dengan WM lain.

OMG, you right. I found the difference after switch from `xfwm4` to `lxqt-kwin-desktop-git`. I will try to fix the problem and report the progress.

**Reply by SannUlya** [reply]
Yep right, thanks man. I fix this problem by install `xfwm4` instead `xfwm4-git`.

**Comment by SannUlya** [comment, marked as answer]
Fixed by installing stable `xfwm4` instead `xfwm4-git`

**Reply by tsujan** [reply]
> Fixed by installing stable xfwm4 instead xfwm4-git

There can be different causes: `xfwm4-git` may have a regression (a bad commit), one of its dependencies may have a problem, or it may need recompilation.

**Comment by furmur** [comment]
https://gitlab.xfce.org/xfce/xfwm4/-/issues/686

**Reply by tsujan** [reply]
Also see this workaround: https://github.com/lxqt/lxqt-panel/pull/1850

**Reply by yan12125** [reply]
Thanka for linking the patch. Actually I consider the change more a fix than a workaround - the current code uses an unneeded flag.

By the way, I'm considering adding that patch to Arch Linux lxqt-panel package (go to [community-staging] first) for better compatibility with xfwm4 4.18.

**Reply by tsujan** [reply]
> Actually I consider the change more a fix than a workaround 

Maybe. It depends on how one sees the new situation.

> By the way, I'm considering adding that patch to Arch Linux lxqt-panel package

Actually, I wanted to ask you whether we should make a point release after merging your PR, or the patch could be applied in Arch.

**Reply by tsujan** [reply]
BTW, have you tested it with Openbox? Most Ubuntu users have it.

**Reply by yan12125** [reply]
> Actually, I wanted to ask you whether we should make a point release after merging your PR, or the patch could be applied in Arch.

A point release sounds better. Patch only the Arch package is also fine.

> BTW, have you tested it with Openbox? Most Ubuntu users have it.

Nope - not sure which WMs are more popular. Will take a look.

**Reply by tsujan** [reply]
> A point release sounds better. 

OK. Will do it next week, in the early days of 2023 and alongside libfm-qt/pcmanfm-qt.

**Reply by tsujan** [reply]
Point releases have been made.

BTW, can we disable CI on pcmanfm-qt? It hasn't been useful so far; on the contrary, its build failures have been annoying.

**Reply by yan12125** [reply]
> BTW, can we disable CI on pcmanfm-qt? It hasn't been useful so far; on the contrary, its build failures have been annoying.

Indeed current CI workflow are not so useful. Here is a short term change: https://github.com/lxqt/pcmanfm-qt/pull/1709. I will post some ideas about a long term solution at https://github.com/lxqt/lxqt/issues/1644.
